kotlin/find-route/src/main/java/com/esri/arcgisruntime/sample/findroute/MainActivity.kt

```diff
@@ -0,0 +1,293 @@
+      Log.e(TAG, error)
+      Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+    }
+  }
+
+  /** Creates a bottom sheet to display a list of direction maneuvers.
+   *
+   * @param directions a list of DirectionManeuver which represents the route
+   */
+  private fun setupBottomSheet(directions: List<DirectionManeuver>) {
+    // create a bottom sheet behavior from the bottom sheet view in the main layout
+    val bottomSheetBehavior = BottomSheetBehavior.from(bottomSheet).apply {
+      // expand the bottom sheet, and ensure it is displayed on the screen when collapsed
+      state = BottomSheetBehavior.STATE_EXPANDED
+      peekHeight = bottomSheet.header.height
+      // animate the arrow when the bottom sheet slides
+      addBottomSheetCallback(object : BottomSheetCallback() {
+        override fun onSlide(bottomSheet: View, slideOffset: Float) {
+          bottomSheet.header.imageView.rotation = slideOffset * 180f
+        }
+
+        override fun onStateChanged(bottomSheet: View, newState: Int) {
+          bottomSheet.header.imageView.rotation = when (newState) {
+            BottomSheetBehavior.STATE_EXPANDED -> 180f
+            else -> 0f
+          }
+        }
+      })
+    }
+
+    bottomSheet.apply {
+      visibility = View.VISIBLE
+      // expand or collapse the bottom sheet when the header is clicked
+      header.setOnClickListener {
+        bottomSheetBehavior.state = when (bottomSheetBehavior.state) {
+          BottomSheetBehavior.STATE_COLLAPSED -> BottomSheetBehavior.STATE_EXPANDED
+          else -> BottomSheetBehavior.STATE_COLLAPSED
+        }
+      }
+      // rotate the arrow so it starts off in the correct rotation
+      header.imageView.rotation = 180f
+
+      directionsListView.apply {
+        // Set the adapter for the list view
+        adapter = ArrayAdapter(
+          this@MainActivity,
+          android.R.layout.simple_list_item_1,
+          directions.map { it.directionText }
+        )
+        // when the user taps a maneuver, set the viewpoint to that portion of the route
+        onItemClickListener =
+          AdapterView.OnItemClickListener { _, _, position, _ ->
+            // remove any graphics that are not the two stops and the route graphic
+            if (graphicsOverlay.graphics.size > 3) {
+              graphicsOverlay.graphics.removeAt(graphicsOverlay.graphics.size - 1)
+            }
+            // set the viewpoint to the selected maneuver
+            val geometry = directions[position].geometry
+            mapView.setViewpointAsync(
+              Viewpoint(geometry.extent, 20.0),
+              1f
+            )
+            // create a graphic with a symbol for the maneuver and add it to the graphics overlay
+            val selectedRouteSymbol = SimpleLineSymbol(
+              SimpleLineSymbol.Style.SOLID,
+              Color.GREEN, 5f
+            )
+            Graphic(geometry, selectedRouteSymbol).also {
```

Comment:
I'd just inline the graphic creation as above

Reply:
I agree, this was before we talked about this pattern!